translations - docs update (#13)
deity-io/falcon-docs: docs/falcon-client/internationalization.md

```text
i18n
  |-en
    |-translations.json
    |-shop.json
    |-blog.json
    |-...
  |-...
```

Default namespace is `translations` and there is also fallback configured to it, in case if translation key can not be found in other namespaces.

HMR for translation files is supported.

## Configuration

Configuration options base on [i18next](https://www.i18next.com/overview/configuration-options) and you can change them via configuration `config.i18n` exported from `bootstrap.js` file:

- `lng: string` - (default: `en`) default application language
- `fallbackLng: string` - (default: `en`) language to use if translations in selected language are not available
- `whitelist: string[]` - (default: `['en']`) available languages, it may be be narrowed, if installed extensions does not support all of specified
- `ns: string[]` - (default: `['translations']`) namespaces used by application
- `debug: boolean` - (default: `false`) i18next debug mode switch
- `resources: object` - (default: `undefined`) allows to inject translations inline, useful during testing

## React components

> Using i18next directly in react is clumsy, so we providing Components to make it simple easy

Reviewer: ```suggestion
Using i18next directly in react is clumsy, so we provide Components to make it very easy
```

### I18n Component

`<I18n>` is a component rendering render props:

- `t: i18next.TranslationFunction` - i18next translation function [see the details](https://www.i18next.com/overview/api#t)
- `i18n: i18next.i18n` - fully initialized i18next instance [see the details](https://www.i18next.com/overview/api#api)

> It is usefully in cases when you would like to have access to `t` function to get translated text inline:

Reviewer: ```suggestion
It is useful in cases when you would like to have an access to `t` function to get a translated text inline:
```

```jsx
<I18n>
  {t => <Image src={items[0].full} alt={t("productGallery.imageAlt")} />}
</I18n>
```

> or you need to have access to `i18next` instance to change the language:

Reviewer: ```suggestion
or you need to have an access to `i18next` instance to change the language:
```

```jsx
<I18n>
  {(_t, i18n) => (
    <LanguageSwitcher
      languages={languages}
      onChange={x => i18n.changeLanguage(x.code)}
    />
  )}
</I18n>
```

### T Component

> As in many places you will need to render translated text as children of other Component, we introduce `<T />` Component which behaves exactly like `i18next.TranslationFunction`. To get translated string for given key you need to use `id` prop instead of `key` as it is reserved keyword in `jsx`

Reviewer: ```suggestion
As in many places you will need to render a translated text as children of other Component, we introduce `<T />` Component which behaves exactly like `i18next.TranslationFunction`. To get translated string for the given key you need to use `id` prop instead of `key` as it is reserved keyword in `jsx`
```

```jsx
<h1>
  <T id="home.title" />
</h1>

// <h1>Home Page<h1>
```

> Others [translationFunction options](https://www.i18next.com/translation-function/essentials#overview-options) are accepted in an unchanged form, for example, to pass

Reviewer: ```suggestion
Other [translationFunction options](https://www.i18next.com/translation-function/essentials#overview-options) are accepted in an unchanged form, for example, to pass
```

```jsx
// product.productsLeftInStock_plural = "{{product}} has {{count}} products left in stock" */

<span>
  <T id="product.productsLeftInStock" product="Zoe Tank" count={3} />
</span>

// -> <span>Zoe Tank has 3 products left in stock<span>
```

### Conventions

Some React Components implement handy conventions to makes translations easier. They also improve the shape of translation resource files as it imposes a certain way of organization. But please bear in mind that none of them are required and there is no need to follow them.

#### Forms

`FormFields` can use its `name` attribute to build translation key for its label or placeholder based on translation context set via `i18nId` property on `Form` component. The rules are:

- `FormField` label - `[Form.i18nId].[FormField.name]FieldLabel`
- `FormField` placeholder - `[Form.i18nId].[FormField.name]FieldPlaceholder`

> In following example, email `FormField` will have automatically translated label if translation for key `signIn.emailFieldLabel` (and placeholder, when `signIn.emailFieldPlaceholder`) will be found.

Reviewer: ```suggestion
In the following example, email `FormField` will have an automatically translated label if a translation for the `signIn.emailFieldLabel` key (and placeholder, when `signIn.emailFieldPlaceholder`) will be found.
```
